swift/ql/lib/codeql/swift/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -399,6 +399,29 @@ private module ReturnNodes {
     override ReturnKind getKind() { result instanceof NormalReturnKind }
   }
 
+  class SelfReturnNode extends ReturnNode, TInoutReturnNode, NodeImpl {
```

Comment:
I'd appreciate a comment on this explaining that it models the `self` value in a constructor / initializer being (implicitly) returned with the newly constructed object.